crates/storage/src/interface.rs

```diff
@@ -606,22 +574,8 @@ impl Interface {
     /// If an error occurs when interacting with the storage system, an error
     /// will be returned.
     ///
-    #[expect(clippy::significant_drop_tightening, reason = "False positive")]
-    pub fn find_by_id<D: Data>(&self, id: Id) -> Result<Option<D>, StorageError> {
-        // TODO: It seems fairly bizarre/unexpected that the put() method is sync
-        // TODO: and not async. The reasons and intentions need checking here, in
-        // TODO: case this find() method should be async and wrap the blocking call
-        // TODO: with spawn_blocking(). However, this is not straightforward at
-        // TODO: present because Slice uses Rc internally for some reason.
-        // TODO: let value = spawn_blocking(|| {
-        // TODO:     self.store.read()
-        // TODO:         .get(&StorageKey::new((*id).into()))
-        // TODO:         .map_err(StorageError::StoreError)
-        // TODO: }).await.map_err(|err| StorageError::DispatchError(err.to_string()))??;
-        let store = self.store.read();
-        let value = store
-            .get(&StorageKey::new(id.into()))
-            .map_err(StorageError::StoreError)?;
+    pub fn find_by_id<D: Data>(id: Id) -> Result<Option<D>, StorageError> {
+        let value = storage_read(id.to_string().as_bytes());
```

Comment:
why `to_string` first?

`storage_read` expects a `&[u8]`, `Id::to_bytes` returns a `&[u8; 16]`

Reply:
The key I added to the store previously allows for 16 - this was added in order to cater for these IDs. (Previously as in, previously merged, in #703).

Reply:
`Id`s are just `Uuid`s which are just `[u8; 16]`, what am I missing?

Reply:
Perhaps I misunderstood the question here - you said:

> storage_read expects a &[u8], Id::to_bytes returns a &[u8; 16]

I thought you were asking about the key type...?

Reply:
I'm saying it can simply be

```suggestion
        let value = storage_read(id.as_bytes());
```

Reply:
Ah, you wrote two separate things. I don't have a problem with the question raised about that. It was the second part with a seemingly-separate observation on the type and/or length that I was replying to.

Reply:
> It was the second part with a seemingly-separate observation on the type and/or length that I was replying to.

that was in reply to yours, indicating that there's no reason why `.to_string()` can't be skipped

> The key I added to the store previously allows for 16 - this was added in order to cater for these IDs. (Previously as in, previously merged, in https://github.com/calimero-network/core/pull/703).

Reply:
Can you reread this thread?

Reply:
There has been zero discussion about `.to_string()` 🙂